### PR TITLE
media-libs/assimp: add 5.2.5-r2, add 5.3.1

### PR DESCRIPTION
**Pull request by negril**
Rework tests to make use of gtest via ctest.

Closes: https://bugs.gentoo.org/908403
Closes: https://bugs.gentoo.org/910362

### CODE REVIEW

File: media-libs/assimp/assimp-5.2.5-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	)
+
+	if use samples; then
+		mycmakeargs+=( -DOpenGL_GL_PREFERENCE="GLVND" )
+	fi
+
+	if use test; then
+		einfo "disabling fma so tests succeed"
+		append-cflags    "-ffp-contract=off"
+		append-cxxpflags "-ffp-contract=off"
```
Comment (thesamesam) [inline media-libs/assimp/assimp-5.2.5-r2.ebuild:79]:
```suggestion
		append-cxxflags "-ffp-contract=off"
```

File: media-libs/assimp/assimp-5.2.5-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		# bug #891787 (CVE-2022-45748), intentionally not in alphabetic ordering
+		-DASSIMP_BUILD_COLLADA_IMPORTER=OFF
+		-DASSIMP_BUILD_COLLADA_EXPORTER=OFF
+	)
+
+	if use samples; then
+		mycmakeargs+=( -DOpenGL_GL_PREFERENCE="GLVND" )
+	fi
+
+	if use test; then
```
Comment (thesamesam) [inline media-libs/assimp/assimp-5.2.5-r2.ebuild:76]: If this is sound to only do for tests, then it needs a comment above it justifying it. But I'd prefer it we just always set it given clearly upstream don't test with contraction.

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: media-libs/assimp/files/assimp-5.2.5-float-comparison.patch [hunk 3]
```diff
@@ -0,0 +1,84 @@
+diff --git a/include/assimp/matrix3x3.inl b/include/assimp/matrix3x3.inl
```
Comment (thesamesam) [inline media-libs/assimp/files/assimp-5.2.5-float-comparison.patch:1]: When you come back to this, please add some explanation to the top. Thanks!

File: media-libs/assimp/assimp-5.3.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+src_test() {
+	local CMAKE_SKIP_TESTS=(
+		# ( Failed )
+		"^utCollada.*"
+		"^utIssues.OpacityBugWhenExporting_727$"
+	)
+	myctestargs+=(
+		--repeat until-pass:100
```
Comment (thesamesam) [inline media-libs/assimp/assimp-5.3.1.ebuild:122]: This deserves a comment above it, please explain why when you come back to this.

File: media-libs/assimp/assimp-5.3.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+		# adds the target headercheck which compiles every header file, default disabled because it adds many targets
+		-DASSIMP_HEADERCHECK=$(usex test)
+	fi
+
+	cmake_src_configure
+}
+
+src_test() {
+	local CMAKE_SKIP_TESTS=(
+		# ( Failed )
```
Comment (thesamesam) [inline media-libs/assimp/assimp-5.3.1.ebuild:117]: Isn't that part obvious? ;)

Comment (negril) [reply media-libs/assimp/assimp-5.3.1.ebuild:117]: The other tests don't timeout or segfault anymore. :]

File: media-libs/assimp/assimp-5.3.1.ebuild (hunk 4 above)
Comment (negril) [reply media-libs/assimp/assimp-5.3.1.ebuild:122]: Will do.
Works together with the better float comparison down below. They generate random input values, so sometimes you gotta find ones that work.

File: media-libs/assimp/files/assimp-5.2.5-float-comparison.patch (hunk 3 above)
Comment (negril) [reply media-libs/assimp/files/assimp-5.2.5-float-comparison.patch:1]: Bit-wise floating point number comparison is down right criminal... It's also a wild mix in there so I just use the operators with tolerances.

**Pull request closed without merging**